src/model/operation/attributeoperation.js

```diff
@@ -107,6 +107,17 @@ export default class AttributeOperation extends Operation {
 		return new AttributeOperation( this.range, this.key, this.newValue, this.oldValue, this.baseVersion + 1 );
 	}
 
+	/**
+	 * @inheritDoc
+	 */
+	toJSON() {
```

Comment:
I wonder here... Theoretically, values could be arrays or objects. Will those be cloned?

Reply:
BTW... maybe we should just give up that and allow only for string, numeral and boolean values for attributes?

Reply:
🤔I'm afraid in a case of objects we pass a reference to the value instead of cloning.
https://github.com/ckeditor/ckeditor5-engine/blob/6c1f87e7e4d8733e9774809374b37db0df04b215/src/model/operation/attributeoperation.js#L98

We can think about dropping other formats than primitives but we don't know if someone has already used it with objects or arrays.